src/core/server/graph/tenant/resolvers/UsernameHistory.ts

```diff
@@ -0,0 +1,16 @@
+import { GQLUsernameHistoryTypeResolver } from "coral-server/graph/tenant/schema/__generated__/types";
+import * as user from "coral-server/models/user";
+
+export const UsernameHistory: Required<
+  GQLUsernameHistoryTypeResolver<user.UsernameHistory>
+> = {
+  createdBy: ({ createdBy }, input, ctx) => {
+    if (createdBy) {
+      return ctx.loaders.Users.user.load(createdBy);
```

Comment:
@wyattjoh, thoughts on ways we can work to clean up this kind of access?

In general, I find if we're being forced to do `something.set.otherSet.item.function(...)` it indicates that the architecture isn't abstracted/decoupled enough for us to easily access the functions/properties we want without digging in really deep.

I worry this will escalate and we'll end up with business logic retrieval that knows too much about the underlying data it's trying to provide when we could maybe abstract this into cleaner retrieval patterns.

Reply:
To me, this reads cleanly. The models being retrieved are nicely written such that reading the chained access tells you what's loading. The last `.load(` even tells me it's a `DataLoader`, which gives me confidence that this call is efficient. We can discuss it in arch meeting today.

Reply:
That's fair, and it probably reads different to you than to me. We can for sure discuss further. 👍